File: 07_reverse_integer/reverse_integer.py
import operator
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 0:
            x = abs(x)
            r = int('-'+str(x)[::-1])
            return r if -2147483648<r<2147483648 else 0
        else:
            r = int(str(x)[::-1])
            return r if -2147483648<r<2147483648 else 0
        

File: 07_reverse_integer/reverse_integer.cpp
class Solution {
public:
    int reverse(int x) {
        long result = 0;
        while(x != 0){
            result = result*10 + x%10;
            x /= 10;
        }
        return (result > INT_MAX || result < INT_MIN)?0:result; // if () return 0, else return result
                
        }
};